src/components/Pages/index.js

```diff
@@ -78,78 +100,76 @@ class PoolDetailPage extends React.Component {
 
   render () {
     let { match, vms, config } = this.props
+
     if (vms.getIn(['pools', match.params.id, 'vm'])) {
       return (<VmDetail vm={vms.getIn(['pools', match.params.id, 'vm'])} pool={vms.getIn(['pools', match.params.id])} config={config} isPool />)
-    } else {
-      if (vms.get('loadInProgress')) {
-        return null
-      }
+    } else if (vms.get('loadInProgress')) {
+      return null
     }
     return <Redirect to='/' />
   }
 }
-
 PoolDetailPage.propTypes = {
   vms: PropTypes.object.isRequired,
   config: PropTypes.object.isRequired,
+
   getPools: PropTypes.func.isRequired,
+
   route: PropTypes.object.isRequired,
   match: PropTypes.object.isRequired,
   location: PropTypes.object.isRequired,
   history: PropTypes.object.isRequired,
   previousPath: PropTypes.string.isRequired,
 }
+const PoolDetailPageConnected = connect(
+  (state) => ({
+    vms: state.vms,
+    config: state.config,
+  }),
+  (dispatch) => ({
+    getPools: ({ poolId }) => dispatch(selectPoolDetail({ poolId })),
+  })
+)(PoolDetailPage)
 
+/**
+ * Route component (for PageRouter) to edit a VM
+ */
 class VmDialogPage extends React.Component {
   componentWillMount () {
+    console.info('VmDialogPage will getCDRom()')
     this.props.getCDRom() // refresh data
-    if (this.props.match.params.id) {
-      this.props.getVms({ vmId: this.props.match.params.id }) // refresh data
 
-      // in case the location is entered from outside
+    // in case the location is entered from outside, refresh data
+    if (this.props.match.params.id) {
+      this.props.getVms({ vmId: this.props.match.params.id })
     }
   }
 
   render () {
     let { match, vms, previousPath } = this.props
     if ((match.params.id && vms.getIn(['vms', match.params.id])) || !match.params.id) {
       return (<VmDialog vm={vms.getIn(['vms', match.params.id])} previousPath={previousPath} />)
+    } else if (vms.get('loadInProgress')) {
+      console.info(`VmDialogPage: VM id can not be found: ${match.params.id}. Load is still in progress - waiting before redirect`)
+      return null
     }
+
+    console.info(`VmDialogPage: VM id can not be found: ${match.params.id}.`)
     return null
   }
 }
-
 VmDialogPage.propTypes = {
   vms: PropTypes.object.isRequired,
+
   getVms: PropTypes.func.isRequired,
   getCDRom: PropTypes.func.isRequired,
```

Comment:
Same.

Reply:
Sure.  Amending the commit now as that small of a change doesn't seem to warrant its own commit.